module-2/backend/Liam_barstad/week_one.txt

```diff
@@ -0,0 +1,92 @@
+List the five common HTTP verbs and what the purpose is of each verb.
+  - get: to fetch views content
+  - post: to upload content and/or add to database
+  - delete: to delete content and/or remove from database
+  - put: ??
+  - update: to rewrite content and/or replace elements in the database
+
+What is Sinatra?
+  Sinatra is a hosting service for ruby-based websites
+
+What is MVC?
+  The model-view-controller style of creating ruby-based websites, where the controller files navigate
+  through webpages, stored in views. It can also add erb content, initiated through models. The models
+  stores and interprets data stored in the database.
+
+Why do we follow conventions when creating our actions/path names in our Sinatra routes?
+  Sinatra looks for specific directories in the program directory, the directory name
+  dictating functionality. However, I don't yet know if websites with different architecture
+  can be hosted through ruby. At very least, action names (and potentially path names) following
+  a specific convention would make websites in Sinatra easy to edit, improve upon, and understand.
+
+What types of variables are accessible in our view templates without explicitly passing them?
+  classes, ids, href values, path values, and parameters are variables that are stored in HTML and CSS.
+  Even tags can be potential variables when values are assigned to groups of them in CSS (maybe class or instance).
```

Comment:
We were thinking of instance variables. You show how to do this in your answer to the next question. Again, I would be careful with language:

* I wouldn't say that parameters are stored in HTML or CSS. `params` is a hash that's created by Sinatra to hold values submitted using query or form params.
* Similarly, I wouldn't say that paths are variables stored in HTML or CSS. The path is part of the HTTP request that allows us to determine what a user has requested.
* I wouldn't consider classes, ids, or href values as variables, but instead as attribute values (see [here](https://www.vsellis.com/anatomy-of-an-html-tag/))